app/GRAPHQL/components/endpointUrl/EndpointURL.tsx

```diff
@@ -0,0 +1,52 @@
+import { encodeBase64 } from '@/app/[...rest]/utils';
+import TextField from '@mui/material/TextField';
+import { useTranslations } from 'next-intl';
+import { usePathname, useSearchParams } from 'next/navigation';
+import { ChangeEvent, FC } from 'react';
+
+type EndpointURLProps = {
+  setURL: (data: string) => void;
+  urlValue: string;
+  setOpen: (data: boolean) => void;
+};
+
+const EndpointURL: FC<EndpointURLProps> = ({ setURL, urlValue, setOpen }) => {
+  const t = useTranslations('apiClient');
+
+  const pathname = usePathname();
+  const pathArray = pathname.split('/');
+  const searchParams = useSearchParams().toString();
+
+  const handleChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setOpen(false);
+
+    const arr = [...pathArray];
+    arr[2] = encodeBase64(encodeURIComponent(value));
+    let newPath = arr.join('/');
+    if (searchParams) {
+      newPath = newPath + `?${searchParams}`;
+    }
+    history.replaceState(null, '', newPath);
+
+    setURL(value);
+  };
+
+  return (
+    <div>
+      <TextField
+        required
+        variant="outlined"
+        label={`${t('urlPlaceholder')}:`}
+        type="text"
+        name="endpointURL"
+        id="endpointURL"
+        value={urlValue}
+        onChange={handleChange}
+        sx={{ width: '100%' }}
+      />
+    </div>
```

Comment:
Do we really have to use `<div> ` wrapper here or maybe `React.Fragment` can be used?

Reply:
Done ${d8065fa14ecad551f9c49312c93d6f38c0eed5c1}